src/Idea.hs

```diff
@@ -79,22 +78,28 @@ showEx tt Idea{..} = unlines $
     f "Found" (Just ideaFrom') ++ f "Perhaps" ideaTo' ++
     ["Note: " ++ n | let n = showNotes ideaNote, n /= ""]
     where
-        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) (second Just.(reduceJunk ideaFrom)) ideaTo
+        (ideaFrom', ideaTo') = maybe (ideaFrom, Nothing) ((second Just) . reduceJunk . (ideaFrom, )) ideaTo
         f msg Nothing = []
         f msg (Just x) | null xs = [msg ++ " you should remove it."]
                        | otherwise = (msg ++ ":") : map ("  "++) xs
             where xs = lines $ tt x
 
-reduceJunk :: String -> String -> (String, String)
-reduceJunk from to =
-    let
-        (tup, rev_tup) = (***) (both lines) (both (reverse.lines)) (dupe (from, to))
+-- traverse through hints output from both direction to take out all the excess 
+reduceJunk :: (String, String) -> (String, String)
+reduceJunk x@(from, to) =
+    let 
+        tup = both lines x; rev_tup = both (reverse. lines) x
+        -- compare each line to get the first dissimilar position
         go :: ([String], [String]) -> Int 
         go ([], _) = 0
         go (_, []) = 0
         go (f : fs, t : ts) = if f == t then 1 + go (fs, ts) else 0
-        (start, end) = ( 1 `max` go tup - 1, 1 `max` go rev_tup)
-    in both (unlines.(\xs-> (xs !!) <$> [start..(length xs - end)])) tup
+        -- subtract 1 in both version to get the corresponding above/below line for context.
+        (start, end) = (go tup - 1, go rev_tup - 1)      
+        -- extract a sub-array from l to r where r is the index in reversed version
+        extract :: [String] -> Int -> Int -> [String]
```

Comment:
A signature like `[a] -> Int -> Int -> [a]`  might be better. In addition, because `drop` and `take` both have the type of `Int -> [a] -> [a]`, you might want to reorder arguments of this function to make its type `Int -> Int -> [a] -> [a]`.